test/base.test.js

```diff
@@ -566,6 +566,133 @@ test('using both policy and all options', async (t) => {
   await t.rejects(app.ready())
 })
 
+test('skip the cache if operation is Subscription', async ({ equal, same, teardown }) => {
+  const app = fastify()
+  teardown(app.close.bind(app))
+  let skipCount = 0
+  let hitCount = 0
+
+  const schema = `
+    type Notification {
+      id: ID!
+      message: String
+    }
+
+    type Query {
+      notifications: [Notification]
+    }
+
+    type Mutation {
+      addNotification(message: String): Notification
+    }
+
+    type Subscription {
+      notificationAdded: Notification
+    }    
+  `
+
+  let idCount = 1
+  const notifications = [
+    {
+      id: idCount,
+      message: 'Notification message'
+    }
+  ]
+
+  const resolvers = {
+    Query: {
+      notifications: () => notifications
+    },
+    Mutation: {
+      addNotification: (_, { message }, { pubsub }) => {
+        const id = idCount++
+        const notification = {
+          id,
+          message
+        }
+        notifications.push(notification)
+
+        pubsub.publish({
+          topic: 'NOTIFICATION_ADDED',
+          payload: {
+            notificationAdded: notification
+          }
+        })
+
+        return notification
+      }
+    },
+    Subscription: {
+      notificationAdded: {
+        subscribe: (_, __, { pubsub }) => pubsub.subscribe('NOTIFICATION_ADDED')
+      }
+    }
+  }
+
+  app.register(mercurius, {
+    schema,
+    resolvers,
+    subscription: true
+  })
+
+  app.register(cache, {
+    all: true,
+    onSkip (type, name) {
+      equal(type, 'Subscription')
+      equal(name, 'notificationAdded')
+      skipCount++
+    },
+    onHit (type, name) {
+      equal(type, 'Subscription')
+      equal(name, 'notificationAdded')
+      hitCount++
+    }
+  })
+
+  const query = 'mutation { addNotification(message: "test") { message } }'
+
+  {
+    const res = await app.inject({
+      method: 'POST',
+      url: '/graphql',
+      body: {
+        query
+      }
+    })
+
+    equal(res.statusCode, 200)
+    same(res.json(), {
+      data: {
+        addNotification: {
+          message: 'test'
+        }
+      }
+    })
+  }
+
+  {
+    const res = await app.inject({
+      method: 'POST',
+      url: '/graphql',
+      body: {
+        query
+      }
+    })
+
+    equal(res.statusCode, 200)
+    same(res.json(), {
+      data: {
+        addNotification: {
+          message: 'test'
+        }
+      }
+    })
+  }
+
+  equal(skipCount, 0)
```

Comment:
@simone-sanfratello Exactly, is not triggered, check this equal, hitCount and hitCount should be 0.

Reply:
could you fix the test accordigly, using `fail` and `plan`?

Reply:
could we also spy the subscription is actually triggered?